File: FirebaseExample/app/src/main/java/com/solie/firebaseexample/MainActivity.java
package com.solie.firebaseexample;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    FirebaseDatabase database = FirebaseDatabase.getInstance();
    DatabaseReference mDatabaseReference;

    EditText ID, Password;
    Button Login, SignUp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ID = findViewById(R.id.login_id);
        Password = findViewById(R.id.login_password);
        Login = findViewById(R.id.login_button);
        SignUp = findViewById(R.id.login_signUp);

        Login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });

    }
}

File: MyApplicationss/app/src/main/java/com/solie/myapplicationss/MainActivity.kt
package com.solie.myapplicationss

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import com.solie.myapplicationss.databinding.ActivityMainBinding
import com.solie.myapplicationss.viewmodel.FirebaseViewModel

class MainActivity : AppCompatActivity() {

    private val fmVmodel : FirebaseViewModel by viewModels()
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        val nameObserver = Observer<String> { newName ->
            binding.textView.text = newName
        }
        fmVmodel.currentName.observe(this, nameObserver)

        binding.button.setOnClickListener {
            val intent = Intent(this, SubActivity::class.java)
            startActivityForResult(intent, 101)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(requestCode == 101) {
            fmVmodel.currentName.setValue(data!!.getStringExtra("name"))
        }
    }

}